index.js

```diff
@@ -185,16 +185,22 @@ function resolveWebpackPath(partial, filename, directory, webpackConfig) {
 
   try {
     var loadedConfig = require(webpackConfig);
-    var aliases = loadedConfig.resolve ? loadedConfig.resolve.alias : [];
-
-    var resolver = webpackResolve.create.sync({
-      alias: aliases
-    });
+    var config = {
+      alias: [],
+    };
+    if (loadedConfig.resolve) {
+      config = {
+        alias: loadedConfig.resolve.alias,
+        extensions: loadedConfig.resolve.extensions,
+      };
+    }
+
+    var resolver = webpackResolve.create.sync(config);
 
     // We don't care about what the loader resolves the partial to
     // we only wnat the path of the resolved file
     partial = stripLoader(partial);
-    var resolvedPath = resolver(directory, partial);
+    var resolvedPath = resolver(path.dirname(filename), partial);
```

Comment:
I understand the case that this change fixes. At the same time, I'd like to make sure it doesn't break any other use cases. I'll need to do some manual tests here and think about how we can create fixture files that would make unit testing this easier.